### DISCUSSION TITLE
Connect-Pnp Online from Azure Automation

### DISCUSSION THREAD

**Discussion by v-nom** [pnp/powershell#1145] [General]
Hi! My team is currently using the legacy version of Pnp Powershell to connect to sharepoint via an Azure Automation Runbook in the cloud. We want to switch to the current version, but are unsure how to do this. Can this be done with Application Permissions? If so, what api permissions will I need for the application?

**Comment by ToddKlindt** [comment]
It's a bit dated, but see if [this article helps](https://pnp.github.io/powershell/articles/azurefunctions.html).

**Comment by AndersRask** [comment]
There isn't much of a difference really, weather you use the old or the new module from Automation Accounts.
You just install the new module, and use what ever client credential or service principal you were using in your runbooks before.

You do need to register the enterprise app and grant it delegate permission as described here https://pnp.github.io/powershell/articles/authentication.html

**Comment by v-nom** [comment]
Okay, here's the part I'm confused on. I cannot complete the documentation below because the "Register-PnPAzureADApp" command prompts me to sign into Azure. The tenant we are using requires a certificate to sign in and the pop up for it does not appear when clicking on it.
[cid:22019ff9-d450-406b-99dd-5d3efe360b8e]

I have created a new application manually, and uploaded a certificate to it. Will that still work?

And what delegate permissions to I need to request for it?
________________________________
